src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -272,7 +325,12 @@ public static function get_theme_data( $deprecated = array(), $options = array()
 	public static function get_block_data() {
 		$registry = WP_Block_Type_Registry::get_instance();
 		$blocks   = $registry->get_all_registered();
-		$config   = array( 'version' => 1 );
+
+		if ( null !== static::$blocks && static::has_same_registered_blocks( 'blocks' ) ) {
+			return static::$blocks;
+		}
+
+		$config = array( 'version' => 2 );
```

Comment:
I've changed version from `1` to `2` because this is a bug I've found while working on this. See https://github.com/WordPress/wordpress-develop/pull/3418/#discussion_r991118993

Reply:
Noting: This change is unrelated to the scope of work for this performance regression. For historical atomic tracking, I'd suggest moving this change to a separate PR or commit. A committer can do this during the commit process if there's not time to remove it in this PR.

Reply:
Let's remove this in the commit. Do it in another commit

Reply:
Committed via changeset https://core.trac.wordpress.org/changeset/54491. I'll remove this change from this PR.